### PR TITLE
Remove ambiguity around uAttributes requirements

### PR DESCRIPTION
**Pull request by sophokles73**
Addresses #71

The requirements regarding the usage of particular UAttributes
properties for the message types supported by uProtocol had been
scattered around multiple documents, some textual descriptions as well
as proto files. It was hard to find the relevant information and to
keep it consistent across the specifications.

The following things have been done to mitigate the situation:
* The uattributes.adoc file is now the single source of truth regarding
  the usage of UAttributes for the supported message types. All
  corresponding definitions have been removed from the cloudevents.adoc
  and uattributes.proto files.
* The cloudevents.adoc file has been moved to the up-l1 folder because
  its content is relevant to uProtocol Client libraries only that use
  CloudEvents' JSON or Protobuf Format to map UMessages to a transport
  protocol's PDU. Application (uEntity) developers will not get in
  touch with CloudEvents at all.

### CODE REVIEW

**Review by PLeVasseur — APPROVED** [review]
Wow @sophokles73, this is great!

Looking forward to your `UUri` cleanup next :slightly_smiling_face:

File: up-l1/cloudevents.adoc [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+
+* MUST be set to `1.0`
+
+
+|`type`
+|https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/spec.md#type[`type`]
+|String
+a|The type of the message
+
+* MUST be set to the value of the `uprotocol.ce_name` option defined for the
```
Comment (stevenhartley) [inline up-l1/cloudevents.adoc:95]:
```suggestion
* *MUST* be set to the value of the `uprotocol.ce_name` option defined for the
```

File: up-l1/cloudevents.adoc [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+link:../up-core-api/uprotocol/uattributes.proto[UMessageType enum].
+
+
+|`id`
+|https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/spec.md#id[`id`]
+|String
+a|Unique identifier for the message
+
+* MUST be set to the https://www.rfc-editor.org/rfc/rfc4122.html#section-3[hyphenated string representation] of the UUID.
```
Comment (stevenhartley) [inline up-l1/cloudevents.adoc:104]:
```suggestion
* *MUST* be set to the https://www.rfc-editor.org/rfc/rfc4122.html#section-3[hyphenated string representation] of the UUID.
```

File: up-l1/cloudevents.adoc [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+
+
+|`source`
+|https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/spec.md#source-1[source]
+|Uri-Reference
+a|The origin (address) of the message
+
+* MUST be set to either the long or short form link:uri.adoc[serialization of the URI]
```
Comment (stevenhartley) [inline up-l1/cloudevents.adoc:112]:
```suggestion
* *MUST* be set to either the long or short form link:uri.adoc[serialization of the URI]
```

File: up-l1/cloudevents.adoc [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+
+
+|`sink`
+|`sink`
+|Uri-Reference
+a|The destination (address) of the message
+
+* MUST be set to either the long or short form link:uri.adoc[serialization of the URI]
```
Comment (stevenhartley) [inline up-l1/cloudevents.adoc:120]:
```suggestion
* *MUST* be set to either the long or short form link:uri.adoc[serialization of the URI]
```

File: up-l1/cloudevents.adoc [hunk 5]
```diff
@@ -0,0 +1,343 @@
...
+
+
+|`priority`
+|`priority`
+|String
+a| The message's priority as defined by link:qos.adoc[QoS].
+
+* MUST be set to the value of the `uprotocol.ce_name` option defined for the
```
Comment (stevenhartley) [inline up-l1/cloudevents.adoc:128]:
```suggestion
* *MUST* be set to the value of the `uprotocol.ce_name` option defined for the
```

File: up-l1/cloudevents.adoc [hunk 6]
```diff
@@ -0,0 +1,343 @@
...
+link:../up-core-api/uprotocol/uattributes.proto[UPriority enum].
+
+
+|`ttl`
+|`ttl`
+|Integer
+a|The amount of time (in milliseconds) after which this message MUST no longer be delivered/processed.
```
Comment (stevenhartley) [inline up-l1/cloudevents.adoc:135]:
```suggestion
a|The amount of time (in milliseconds) after which this message *MUST* no longer be delivered/processed.
```

File: up-l1/cloudevents.adoc [hunk 7]
```diff
@@ -0,0 +1,343 @@
...
+|Integer
+|The service consumer's permission level as defined in link:../up-l2/permissions.adoc#_code_based_access_permissions_caps[Code-Based uE Access Permissions (CAPs)]
+
+
+|`reqid`
+|`reqid`
+|String 
+a|The identifier that a service consumer can use to correlate an RPC Repsonse message with its RPC Request.
+
+* MUST be set to the https://www.rfc-editor.org/rfc/rfc4122.html#section-3[hyphenated string representation] of the UUID.
```
Comment (stevenhartley) [inline up-l1/cloudevents.adoc:154]:
```suggestion
* *MUST* be set to the https://www.rfc-editor.org/rfc/rfc4122.html#section-3[hyphenated string representation] of the UUID.
```

File: up-l1/cloudevents.adoc [hunk 8]
```diff
@@ -0,0 +1,343 @@
...
+
+|`commstatus`
+|`commstatus`
+|Integer
+a|A code indicating an error that has occurred during the delivery of either an RPC Request or Response message.
+
+* MUST be set to the link:../up-core-api/uprotocol/ustatus.proto[UCode]'s integer value
```
Comment (stevenhartley) [inline up-l1/cloudevents.adoc:161]:
```suggestion
* *MUST* be set to the link:../up-core-api/uprotocol/ustatus.proto[UCode]'s integer value
```

File: up-l1/cloudevents.adoc [hunk 9]
```diff
@@ -0,0 +1,343 @@
...
+|===
+
+[#ce-formats]
+=== uPayload Mapping
+
+The sections below define the mapping of uPayload to CloudEvent attributes.
+
+==== Mapping to CloudEvent Protobuf Format
+
+The rules defined in https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/formats/protobuf-format.md[Protobuf Event Format for CloudEvents, Version 1.0.2] MUST be applied when mapping UPayload to CloudEvents using the Protobuf Format. The table below defines specific values to use for the different UMessage payload types.
```
Comment (stevenhartley) [inline up-l1/cloudevents.adoc:177]:
```suggestion
The rules defined in https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/formats/protobuf-format.md[Protobuf Event Format for CloudEvents, Version 1.0.2] *MUST* be applied when mapping UPayload to CloudEvents using the Protobuf Format. The table below defines specific values to use for the different UMessage payload types.
```

File: up-l1/cloudevents.adoc [hunk 10]
```diff
@@ -0,0 +1,343 @@
...
+
+This mapping MUST be used by link:../upclient.adoc[uProtocol Client Libraries] which use the Protobuf Event Format for the binding to a particular transport protocol.
```
Comment (stevenhartley) [inline up-l1/cloudevents.adoc:179]:
```suggestion
This mapping *MUST* be used by link:../upclient.adoc[uProtocol Client Libraries] which use the Protobuf Event Format for the binding to a particular transport protocol.
```

File: up-l1/cloudevents.adoc [hunk 11]
```diff
@@ -0,0 +1,343 @@
...
+|`text/plain`
+|*MAY* be set to a URI-Reference identifying the schema that the data adheres to
+|`text_data`
+
+|===
+
+
+==== Mapping to CloudEvent JSON Format
+
+The rules defined in https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/formats/json-format.md[JSON Event Format for CloudEvents, Version 1.0.2] MUST be applied when mapping UPayload to CloudEvents using the JSON Format. The table below defines specific values to use for the different UMessage payload types.
```
Comment (stevenhartley) [inline up-l1/cloudevents.adoc:225]:
```suggestion
The rules defined in https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/formats/json-format.md[JSON Event Format for CloudEvents, Version 1.0.2] *MUST* be applied when mapping UPayload to CloudEvents using the JSON Format. The table below defines specific values to use for the different UMessage payload types.
```

File: up-l1/cloudevents.adoc [hunk 12]
```diff
@@ -0,0 +1,343 @@
...
+
+This mapping MUST be used by link:../upclient.adoc[uProtocol Client Libraries] which use the JSON Event Format for the binding to a particular transport protocol.
```
Comment (stevenhartley) [inline up-l1/cloudevents.adoc:227]:
```suggestion
This mapping *MUST* be used by link:../upclient.adoc[uProtocol Client Libraries] which use the JSON Event Format for the binding to a particular transport protocol.
```

**Review by stevenhartley — CHANGES_REQUESTED** [review]
Very minor changes but thankyou so much @sophokles73 !!

File: basics/cloudevents.adoc
[the remark this answers is not in the document]
Comment (sophokles73) [reply basics/cloudevents.adoc:1]: Good point, I think that should be doable.

**Comment by sophokles73** [comment]
@stevenhartley I have fixed the formatting and used `git mv` to move the cloudevents stuff. Can you check if this did the trick for you? I can then squash and merge ...

File: up-l1/cloudevents.adoc [hunk 13]
```diff
@@ -0,0 +1,343 @@
+= CloudEvents
+:toc:
+:sectnums:
+
+The key words "*MUST*", "*MUST NOT*", "*REQUIRED*", "*SHALL*", "*SHALL NOT*", "*SHOULD*", "*SHOULD NOT*", "*RECOMMENDED*", "*MAY*", and "*OPTIONAL*" in this document are to be interpreted as described in https://www.rfc-editor.org/info/bcp14[IETF BCP14 (RFC2119 & RFC8174)]
+
+----
+SPDX-FileCopyrightText: 2023 Contributors to the Eclipse Foundation
```
[the remark this answers is not in the document]
Comment (sophokles73) [reply up-l1/cloudevents.adoc:8]: I do not see why it shouldn't. I wanted to remove the duplication of the copyright info. Do you have any particular concerns?

[the remark this answers is not in the document]
Comment (stevenhartley) [reply up-l1/cloudevents.adoc:8]: nope, it is fine

**Comment by stevenhartley** [comment]
> the cloudevents stuff. Can you check if this di

Didn't seem to work (git mv) but that is fine, I'll just look at the history if needed

**Review by stevenhartley — APPROVED** [review]

**Pull request merged**